services/ble/Gap.cpp

```diff
@@ -207,9 +225,59 @@ namespace services
             return localName;
     }
 
-    infra::ConstByteRange GapAdvertisingDataParser::ManufacturerSpecificData() const
+    infra::Optional<std::pair<uint16_t, infra::ConstByteRange>> GapAdvertisingDataParser::ManufacturerSpecificData() const
+    {
+        auto data = ParserAdvertisingData(GapAdvertisementDataType::manufacturerSpecificData);
+
+        if (data.empty())
+            return infra::none;
+
+        if (data.size() < sizeof(uint16_t))
+            return infra::none;
+
+        auto manufacturerCode = infra::ReinterpretCastMemoryRange<const uint16_t>(infra::Head(data, sizeof(uint16_t))).front();
+        auto manufacturerData = infra::DiscardHead(data, sizeof(uint16_t));
+
+        return infra::MakeOptional(std::make_pair(manufacturerCode, manufacturerData));
+    }
+
+    infra::Optional<GapPeripheral::AdvertisementFlags> GapAdvertisingDataParser::Flags() const
+    {
+        auto flagsData = ParserAdvertisingData(GapAdvertisementDataType::flags);
+
+        if (flagsData.empty())
+            return infra::none;
+
+        if (flagsData.size() != 1)
+            return infra::none;
+
+        return infra::MakeOptional(static_cast<GapPeripheral::AdvertisementFlags>(flagsData[0]));
+    }
+
+    infra::MemoryRange<AttAttribute::Uuid16> GapAdvertisingDataParser::CompleteListOf16BitUuids() const
+    {
+        auto uuidData = ParserAdvertisingData(GapAdvertisementDataType::completeListOf16BitUuids);
+
+        if (uuidData.empty())
+            return infra::MemoryRange<AttAttribute::Uuid16>();
+
+        if (uuidData.size() % sizeof(AttAttribute::Uuid16) != 0)
+            return infra::MemoryRange<AttAttribute::Uuid16>();
+
+        return infra::ConstCastMemoryRange<AttAttribute::Uuid16>(infra::ReinterpretCastMemoryRange<const AttAttribute::Uuid16>(uuidData));
+    }
+
+    infra::MemoryRange<AttAttribute::Uuid128> GapAdvertisingDataParser::CompleteListOf128BitUuids() const

```

Comment:
Why doesn't this function return infra::MemoryRange<const AttAttribute::Uuid128>?
(Similar question for ComplteeListOf16BitUuids)